worker/agentconfigupdater/manifold.go

```diff
@@ -81,11 +102,20 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 					info.PrivateKey = existing.PrivateKey
 				}
 				config.SetStateServingInfo(info)
+				if needsRestart {
```

Comment:
"needsRestart" is a little misleading as it's really "mongoMemoryProfileChanged".
That change would make the code here make more sense since on the surface there's no relation between needing a restart (in general), and writing a new memory profile setting.

There could be a needsRestart variable as well for use below. Currently it would just be 
```
needsRestart := mongoMemoryProfileChanged
```
but maybe more conditions could be added later.

Reply:
Fair enough.